graylog2-web-interface/src/components/common/SourceCodeEditor.md

```diff
@@ -0,0 +1,67 @@
+Light-themed editor:
+```js
+const createReactClass = require('create-react-class');
+
+const LightSourceEditor = createReactClass({
+  getInitialState() {
+    return {
+      code: `function foobar() {
+  console.log('this is some source code!');
+}
+`,
+    };
+  },
+  
+  handleChange(nextValue) {
+    this.setState({ code: nextValue });
+  },
+  
+  render() {
+    const { code } = this.state;
+    return (
+      <div>
+        <SourceCodeEditor id="editor-1"
+                          theme="light"
+                          value={code}
+                          onChange={this.handleChange} />
+        <p>Preview:</p>
+        <pre>{code}</pre>
+      </div>
+    );
+  },
+});
+
+<LightSourceEditor />
+```
+
+Read-only dark-themed editor:
+```js
+const code = `function foobar() {
+  console.log('this is some source code!');
+}
+`;
+const annotations = [
+  { row: 1, column: -1, text: 'oh noes!', type: 'error' },
+  { row: 2, column: -1, text: 'easy!', type: 'warning' },
+  { row: 3, column: -1, text: 'info!', type: 'info' },
+];
+<SourceCodeEditor id="editor-2"
+                  annotations={annotations}
+                  readOnly
+                  theme="dark"
+                  value={code} />
+```
+
+Non-resizable editor without toolbar and with custom height and width:
```

Comment:
States that the editor isn't resizable, when in fact it is.

Reply:
That was me changing props and texts last minute 😅 Will fix it.